src/Strategy/Haskell/Cabal.hs

```diff
@@ -109,47 +109,47 @@ cabalGenPlanCmd =
 cabalPlanFilePath :: Path Rel File
 cabalPlanFilePath = $(mkRelFile "dist-newstyle/cache/plan.json")
 
-isCabalFile :: Path Abs File -> Bool
-isCabalFile file = isDotCabal || isCabalDotProject
-  where
-    name = fileName file
-    isDotCabal = ".cabal" `isSuffixOf` name
-    isCabalDotProject = "cabal.project" == name
-
 discover :: (Has ReadFS sig m, Has Diagnostics sig m, Has ReadFS rsig run, Has Exec rsig run, Has Diagnostics rsig run) => Path Abs Dir -> m [DiscoveredProject run]
 discover dir = context "Cabal" $ do
   projects <- context "Finding projects" $ findProjects dir
   pure (map mkProject projects)
 
+isDotCabal :: Path Abs File -> Bool
+isDotCabal name = ".cabal" `isSuffixOf` fileName name
+
+isCabalDotFile :: Path Abs File -> Bool
+isCabalDotFile name = "cabal.project" == fileName name
+
 findProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [CabalProject]
 findProjects = walk' $ \dir _ files -> do
-  let project =
-        CabalProject
-          { cabalDir = dir
-          }
-
-  if any isCabalFile files
-    then pure ([project], WalkSkipAll)
-    else pure ([], WalkContinue)
+  -- TODO: @wes is this the right approach?
```

Comment:
@scruffystuffs - question for you since it looks like you may have built this. If there is both a *.cabal file AND a cabal.project file, I assume we should just be setting the origin path here to the *.cabal file (since we specify our deps in the spectrometer.cabal file. Wanted to confirm this behavior with you.

For more context, this field will be used to signal to users where we found deps (i.e. what manifest file)

Reply:
I believe that's correct.  @cnr actually knows cabal better than me, so I'd double-check with him instead.

Reply:
it's important that we continue to look for `cabal.project`. cabal supports multi-module projects, where `cabal.project` denotes the root of the project, and subdirectories can contain `.cabal` files

notably the folder containing `cabal.project` is **not required to contain a `whatever.cabal` file**

the short-term less-accurate version here is to use the whole list from `filter isCabalFile` as manifest files in the project

the long-term more-accurate version is parse `cabal.project` and look for relevant cabal files to mark as manifests

Reply:
sounds good, I've updated it to use the short-term less-accurate version